org.mwc.debrief.legacy/src/Debrief/ReaderWriter/Nisida/ImportNisida.java

```diff
@@ -0,0 +1,576 @@
+			return lastEntryWithText;
+		}
+
+		public void setLastEntryWithText(String lastEntryWithText) {
+			this.lastEntryWithText = lastEntryWithText;
+		}
+
+		public int getMonth() {
+			return month;
+		}
+
+		public void setMonth(int month) {
+			this.month = month;
+		}
+
+		public int getYear() {
+			return year;
+		}
+
+		public void setYear(int year) {
+			this.year = year;
+		}
+
+		public TrackWrapper getPlatform() {
+			return platform;
+		}
+
+		public void setPlatform(TrackWrapper platform) {
+			this.platform = platform;
+		}
+
+		public List<ImportNisidaError> getErrors() {
+			return errors;
+		}
+
+		public void setErrors(List<ImportNisidaError> errors) {
+			this.errors = errors;
+		}
+
+		public Date getTimestamp() {
+			return timestamp;
+		}
+
+		public void setTimestamp(Date timestamp) {
+			this.timestamp = timestamp;
+		}
+
+		public int getLineNumber() {
+			return lineNumber;
+		}
+
+		public void setLineNumber(int lineNumber) {
+			this.lineNumber = lineNumber;
+		}
+
+		public Layers getLayers() {
+			return layers;
+		}
+
+	}
+
+	/**
+	 * Nisida Importer
+	 */
+	public ImportNisida() {
+
+	}
+
+	public static boolean canLoadThisFile(final InputStream is) {
+		final BufferedReader br = new BufferedReader(new InputStreamReader(is));
+
+		try {
+			String nisidaLine;
+			while ((nisidaLine = br.readLine()) != null) {
+				if (nisidaLine != null && nisidaLine.startsWith("UNIT/")) {
+					return true;
+				}
+			}
+		} catch (IOException e) {
+			// There were problems reading the file. It cannot be loaded.
+		}
+		return false;
+	}
+
+	/**
+	 * Method that should be called to load the NISIDA file
+	 * 
+	 * @param is
+	 */
+	public static void importThis(final InputStream is, final Layers layers) {
+		final BufferedReader br = new BufferedReader(new InputStreamReader(is));
+
+		try {
+			String nisidaLine;
+			int lineNumber = 1;
+			final NisidaLoadState status = new NisidaLoadState(layers);
+			while ((nisidaLine = br.readLine()) != null) {
+				status.setLineNumber(lineNumber);
+				loadThisLine(nisidaLine, status);
+				++lineNumber;
+			}
+		} catch (IOException e) {
+			// There were problems reading the file. It cannot be loaded.
+		}
+	}
+
+	private static void loadThisLine(final String line, final NisidaLoadState status) {
+		if (line.startsWith("UNIT/")) {
+			/**
+			 * Handle UNIT line giving month, year and platform Format is:
+			 * UNIT/ADRI/OCT03/SRF/
+			 */
+			final String[] tokens = line.split("/");
+
+			final String platformName = tokens[1];
+
+			// FIND THE PLATFORM.
+			TrackWrapper track = (TrackWrapper) status.getLayers().findLayer(platformName);
+			if (track == null) {
+				track = new TrackWrapper();
+				track.setColor(DebriefColors.RED);
+				track.setName(platformName);
+				status.getLayers().addThisLayer(track);
+			}
+			status.setPlatform(track);
+
+			final String dateString = tokens[2];
+			final DateFormat dateFormatter = new SimpleDateFormat("MMMyy");
```

Comment:
@saulhidalgoaular - this looks like unnecessary object creation `SimpleDateFormat` we should do it in the parent class, and just pass it in as a parameter

Reply:
https://github.com/debrief/debrief/pull/4905/commits/1fa769ea6d54c4c137a9dfe4761c7740a6e8aa3b